Added a features section the readme (#1560)
dotnet/interactive: README.md

* [Code bots](https://github.com/CodeConversations/CodeConversations)
* Devices like [Raspberry Pi](https://www.raspberrypi.org/)
* Embeddable script engines

> * [REPLs](https://github.com/jonsequitur/dotnet-repl)

Reviewer: 😊

Author: Should we add a section on it?

Reviewer: No strong opinion.

We recently introduced the [.NET Interactive Notebooks](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode) extension for Visual Studio Code, which adds support for .NET Interactive using the new Visual Studio Code [native notebook feature](https://code.visualstudio.com/updates/v1_45#_github-issue-notebook). We encourage you to [try it out](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode). If you'd like to contribute, you can start [here](CONTRIBUTING.md). 

![newnotebook](https://user-images.githubusercontent.com/2546640/130660742-deb5c33b-020d-4d03-8034-7f11532c3201.gif)

---

## Features
### Multi-language notebooks 
.NET Interactive enables users to mix languages in a single notebook or cell without a wrapper. The multi-language experience opens up doors for users to use the best language for the task at hand.

**Languages supported**
- C# and F# 
- PowerShell built in collaboration with the PowerShell team 💙
- JavaScript
- HTML 
- SQL built in collaboration with the  Azure Data / SQL team 💙

**Coming soon**
- Kusto 

> **What languages are we exploring** 

Reviewer: ```suggestion
**What languages are we exploring?** 
```

- Python and R 

### Variable Sharing 

.NET Interactive enables you to write code in multiple languages within a single notebook and in order to take advantage of those languages' different strengths, you might find it useful to share data between them. Read more [here](https://github.com/dotnet/interactive/blob/main/docs/variable-sharing.md).

![Notebooks-variable-sharing](https://user-images.githubusercontent.com/2546640/130664292-1cdfb806-a6f6-4874-bcad-a5eb4517a925.gif)

The gif above showcases the following:

> - Variable sharing across C#, HTML and JavaScript across cells.

Reviewer: ```suggestion
- Variable sharing across C#, HTML and JavaScript cells.
```

- Multi-language cells.

> For more examples on multi-language notebooks and variable sharing checkout our [polyglot samples](https://github.com/dotnet/interactive/tree/main/samples/notebooks/polyglot).

Reviewer: ```suggestion
For more examples on multi-language notebooks and variable sharing check out our [polyglot samples](https://github.com/dotnet/interactive/tree/main/samples/notebooks/polyglot).
```

### Visualization 

**Low code visualization**

In just a single line of code easily visualize data with Microsoft SandDance and [nteract DataExplorer](https://data-explorer.nteract.io/). For  example the code snippet below will render an interactive [Microsoft SandDance](https://www.microsoft.com/en-us/research/project/sanddance/).
```csharp
housingData.ExploreWithSandDance().Display();
```

![low-code-visualization](https://user-images.githubusercontent.com/2546640/130510820-6a5b5f9d-a0cc-4fef-8a3d-ea741a30d7f8.gif)

> For more [low code visualization](https://github.com/dotnet/interactive/tree/main/samples/ExtensionLab) example check out our samples.

Reviewer: ```suggestion
For more [low code visualization](https://github.com/dotnet/interactive/tree/main/samples/ExtensionLab) examples, check out our samples.
```
